lms/services/lti_grading/_v13.py

```diff
@@ -78,20 +78,16 @@ def record_result(self, grading_id, score=None, pre_record_hook=None):
                 # We silently shallow this type of error
                 return None
 
-            if (
-                # Blackboard
-                (
-                    err.status_code == 400
-                    and "User could not be found:" in err.response.text
-                )
-                # D2L
-                or (
-                    err.status_code == 403
-                    and "User in requested score is not enrolled in the org unit"
-                    in err.response.text
-                )
-            ):
-                raise StudentNotInCourse(grading_id) from err
+            for expected_code, expected_text in [
+                (400, "User could not be found:"),
+                (403, "User in requested score is not enrolled in the org unit"),
+                (404, "User in requested score does not exist"),
```

Comment:
This one is new and discovered as part of the testing instructions.

Reply:
Might be good to keep the comments saying where each error message was discovered